pkg/testing/testmetrics/testexporter.go

```diff
@@ -15,87 +15,280 @@
-func diff(got, want []*view.Row) string {
-	for i := 0; i < len(got); i++ {
-		if i >= len(want) {
-			break
+// convertMetricToSimpleFormat converts OpenTelemetry metrics to simple MetricData
+func (e *TestExporter) convertMetricToSimpleFormat(metric metricdata.Metrics) {
+	switch data := metric.Data.(type) {
+	case metricdata.Sum[int64]:
+		for _, point := range data.DataPoints {
+			e.metrics = append(e.metrics, MetricData{
+				Name:   metric.Name,
+				Value:  float64(point.Value),
+				Labels: e.attributesToMap(point.Attributes),
+			})
+		}
+	case metricdata.Sum[float64]:
+		for _, point := range data.DataPoints {
+			e.metrics = append(e.metrics, MetricData{
+				Name:   metric.Name,
+				Value:  point.Value,
+				Labels: e.attributesToMap(point.Attributes),
+			})
 		}
-		if got[i] == want[i] {
-			continue
+	case metricdata.Gauge[int64]:
+		for _, point := range data.DataPoints {
+			e.metrics = append(e.metrics, MetricData{
+				Name:   metric.Name,
+				Value:  float64(point.Value),
+				Labels: e.attributesToMap(point.Attributes),
+			})
 		}
-		if len(got[i].Tags) > 0 && len(want[i].Tags) > 0 && !reflect.DeepEqual(got[i].Tags, want[i].Tags) {
-			return fmt.Sprintf("Expected metric tags not found, -want, +got:\n- %s\n+ %s",
-				want[i].Tags, got[i].Tags)
+	case metricdata.Gauge[float64]:
+		for _, point := range data.DataPoints {
+			e.metrics = append(e.metrics, MetricData{
+				Name:   metric.Name,
+				Value:  point.Value,
+				Labels: e.attributesToMap(point.Attributes),
+			})
 		}
-		if !cmp.Equal(got[i].Data, want[i].Data, cmpopts.IgnoreTypes(time.Time{})) {
-			return fmt.Sprintf("Expected metric value not found, -want, +got:\n- %s\n+ %s",
-				want[i].Data, got[i].Data)
+	case metricdata.Histogram[float64]:
+		for _, point := range data.DataPoints {
+			e.metrics = append(e.metrics, MetricData{
+				Name:   metric.Name,
+				Value:  point.Sum,
+				Labels: e.attributesToMap(point.Attributes),
+			})
 		}
+	case metricdata.Histogram[int64]:
+		for _, point := range data.DataPoints {
+			e.metrics = append(e.metrics, MetricData{
+				Name:   metric.Name,
+				Value:  float64(point.Sum),
+				Labels: e.attributesToMap(point.Attributes),
+			})
+		}
+	// Add more cases as needed for other metric types
+	default:
+		fmt.Printf("Warning: Unsupported metric type: %T\n", data)
+	}
+}
+
+// attributesToMap converts OpenTelemetry attributes to a simple map
+func (e *TestExporter) attributesToMap(attrs attribute.Set) map[string]string {
+	result := make(map[string]string)
+	iter := attrs.Iter()
+	for iter.Next() {
+		kv := iter.Attribute()
+		result[string(kv.Key)] = kv.Value.AsString()
 	}
-	if len(got) > len(want) {
-		var sb strings.Builder
-		sb.WriteString("Unexpected metric(s) found:\n")
-		for _, row := range got[len(want):] {
-			sb.WriteString("+ ")
-			sb.WriteString(row.String())
+	return result
+}
+
+// Global test reader to capture metrics
+var globalTestReader sdkmetric.Reader
```

Comment:
Is the global reader necessary? The previous scaffolding seems to have been able to do this without a global

Reply:
testexporter.go is refactored in https://github.com/GoogleContainerTools/config-sync/pull/1909 and global reader is no longer needed.